src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file radial_basis_function_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "radial_basis_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType,
+         typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+         typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF(
+    const size_t inSize,
+    const size_t outSize,
+    arma::mat& centres) :
+    inSize(inSize),
+    outSize(outSize),
+    centres(centres)
+{
+  sigmas = arma::ones(outSize, 1);
+}
+
+template<typename InputDataType, typename OutputDataType,
+         typename Activation>
+template<typename eT>
+void RBF<InputDataType, OutputDataType, Activation>::Forward(
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
+{
+  distances = arma::mat(outSize, input.n_cols);
+  arma::mat betas;
+
+  for (size_t i = 0; i < input.n_cols; i++)
+  {
+    arma::mat temp = centres.each_col() - input.col(i);
+    distances.col(i) = arma::pow(arma::sum(
+                                 arma::pow((temp),
+                                 2), 0), 0.5).t();
+  }
+  if(input.n_cols > 1)
+  {
+    if (sigmas(0) == 1)
+    {
+      sigmas = arma::mean(distances, 1);
+    }
+    else
+    {
+      sigmas = (sigmas + arma::mean(distances, 1))/2;
+    }
+  }
```

Comment:
Hey could you provide the link to the paper /implementation you are referring to for calculating these sigma values?

Reply:
https://link.springer.com/chapter/10.1007/978-3-319-19620-6_27 . This might help

Reply:
Actually it is a pretty versatile technique that's why too many experiments have done on this.